src/toil/jobStores/aws/jobStore.py

```diff
@@ -815,8 +863,10 @@ def bucket_retry_predicate(error):
 
                 return bucket
 
-    def _bindDomain(self, domain_name, create=False, block=True):
+    def _bindDomain(self, domain_name: str, create: bool = False, block: bool = True) -> Optional[str]:
```

Comment:
Since there's no domain object anymore, this now returns a string that you passed it to indicate success, or None to indicate failure.

That's not a design that we would have arrived at if we had set out to build this on Boto 3 to start with.

Can we get it to not return anything and raise an exception if create=False and it doesn't exist? And change the usages to work with that?